File: ChatDemo-UI3.0/ChatDemo-UI3.0/Code/Utilities/EMSDKHelper.m
/************************************************************
 *  * Hyphenate
 * __________________
 * Copyright (C) 2016 Hyphenate Inc. All rights reserved.
 *
 * NOTICE: All information contained herein is, and remains
 * the property of Hyphenate Inc.
 */

#import "EMSDKHelper.h"

#import "EMNotificationNames.h"

@implementation EMSDKHelper

+ (EMMessage *)sendTextMessage:(NSString *)text
                            to:(NSString *)receiver
                      chatType:(EMChatType)chatType
                    messageExt:(NSDictionary *)messageExt

{
    EMTextMessageBody *body = [[EMTextMessageBody alloc] initWithText:text];

    NSString *sender = [[EMClient sharedClient] currentUsername];
    EMMessage *message = [[EMMessage alloc] initWithConversationID:receiver
                                                              from:sender
                                                                to:receiver
                                                              body:body
                                                               ext:messageExt];
    message.chatType = chatType;
    
    return message;
}

+ (EMMessage *)sendCmdMessage:(NSString *)action
                           to:(NSString *)receiver
                     chatType:(EMChatType)chatType
                   messageExt:(NSDictionary *)messageExt
                    cmdParams:(NSArray *)params
{
    EMCmdMessageBody *body = [[EMCmdMessageBody alloc] initWithAction:action];
    
    if (params) {
        body.params = params;
    }

    NSString *sender = [[EMClient sharedClient] currentUsername];
    EMMessage *message = [[EMMessage alloc] initWithConversationID:receiver
                                                              from:sender
                                                                to:receiver
                                                              body:body
                                                               ext:messageExt];
    message.chatType = chatType;
    
    return message;
}

+ (EMMessage *)sendLocationMessageWithLatitude:(double)latitude
                                     longitude:(double)longitude
                                       address:(NSString *)address
                                            to:(NSString *)receiver
                                      chatType:(EMChatType)chatType
                                    messageExt:(NSDictionary *)messageExt
{
    EMLocationMessageBody *body = [[EMLocationMessageBody alloc] initWithLatitude:latitude
                                                                        longitude:longitude
                                                                          address:address];

    NSString *sender = [[EMClient sharedClient] currentUsername];
    EMMessage *message = [[EMMessage alloc] initWithConversationID:receiver
                                                              from:sender
                                                                to:receiver
                                                              body:body
                                                               ext:messageExt];
    message.chatType = chatType;
    
    return message;
}

+ (EMMessage *)sendImageData:(NSData *)imageData
                 displayName:(NSString *)displayName
                          to:(NSString *)receiver
                    chatType:(EMChatType)chatType
                  messageExt:(NSDictionary *)messageExt
{
    EMImageMessageBody *body = [[EMImageMessageBody alloc] initWithData:imageData displayName:displayName];
    
    if (CGSizeEqualToSize(body.size, CGSizeZero)) {
        if (imageData.length) {
            UIImage *image = [UIImage imageWithData:imageData];
            body.size = image.size;
        }
    }
    
    NSString *sender = [[EMClient sharedClient] currentUsername];
    EMMessage *message = [[EMMessage alloc] initWithConversationID:receiver
                                                              from:sender
                                                                to:receiver
                                                              body:body
                                                               ext:messageExt];
    
    message.chatType = chatType;
    
    return message;
}

+ (EMMessage *)sendVoiceMessageWithLocalPath:(NSString *)localPath
                                 displayName:(NSString *)displayName
                                    duration:(NSInteger)duration
                                          to:(NSString *)receiver
                                    chatType:(EMChatType)chatType
                                  messageExt:(NSDictionary *)messageExt
{
    EMVoiceMessageBody *body = [[EMVoiceMessageBody alloc] initWithLocalPath:localPath displayName:displayName];
    
    if (duration > 0) {
        body.duration = (int)duration;
    }
    
    NSString *sender = [[EMClient sharedClient] currentUsername];
    EMMessage *message = [[EMMessage alloc] initWithConversationID:receiver
                                                              from:sender
                                                                to:receiver
                                                              body:body
                                                               ext:messageExt];
    message.chatType = chatType;
    
    return message;
}

+ (EMMessage *)sendVideoMessageWithLocalURL:(NSURL *)url
                                displayName:(NSString *)displayName
                                   duration:(NSInteger)duration
                                         to:(NSString *)receiver
                                   chatType:(EMChatType)chatType
                                 messageExt:(NSDictionary *)messageExt
{
    EMVideoMessageBody *body = [[EMVideoMessageBody alloc] initWithLocalPath:[url path] displayName:displayName];
    
    if (duration > 0) {
        body.duration = (int)duration;
    }
    
    NSString *sender = [[EMClient sharedClient] currentUsername];
    EMMessage *message = [[EMMessage alloc] initWithConversationID:receiver
                                                              from:sender
                                                                to:receiver
                                                              body:body
                                                               ext:messageExt];
    message.chatType = chatType;
    
    return message;
}

@end

File: swift-demo/Hyphenate-Demo-Swift/Code/Utilities/EMSDKHelper.swift
//
//  EMSDKHelper.swift
//  Hyphenate-Demo-Swift
//
//  Created by dujiepeng on 2017/6/22.
//  Copyright © 2017 dujiepeng. All rights reserved.
//

import UIKit
import Hyphenate

let sender = EMClient.shared().currentUsername

class EMSDKHelper: NSObject {
    class func createTextMessage(_ text: String, to receiver: String, _ chatType: EMChatType, _ ext: Dictionary<String, Any>?) -> EMMessage{
        let body = EMTextMessageBody.init(text: text)
        let msg = EMMessage.init(conversationID: receiver, from: sender, to: receiver, body: body, ext: ext)
        msg!.chatType = chatType
        return msg!
    }
    
    class func createCMDMessage(_ action: String, to receiver: String, _ chatType: EMChatType, _ ext: Dictionary<String, Any>?, _ cmdParams: Dictionary<String, Any>?) -> EMMessage{
        let body = EMCmdMessageBody.init(action: action)
        let msg = EMMessage.init(conversationID: receiver, from: sender, to: receiver, body: body, ext: ext)
        msg!.chatType = chatType
        return msg!
    }
    
    class func createLocationMessage(_ latitude: Double, _ longitude: Double, _ address: String , to receiver: String, _ chatType: EMChatType, _ ext: Dictionary<String, Any>?) -> EMMessage {
        let body = EMLocationMessageBody.init(latitude: latitude, longitude: longitude, address: address)
        let msg = EMMessage.init(conversationID: receiver, from: sender, to: receiver, body: body, ext: ext)
        msg!.chatType = chatType
        return msg!
    }
    
    class func createImageMessage(_ image: UIImage, _ displayName: String, to receiver: String, _ chatType: EMChatType, _ ext: Dictionary<String, Any>?) -> EMMessage {
        let data = UIImageJPEGRepresentation(image, 1)
        let body = EMImageMessageBody.init(data: data, displayName: displayName)
        body?.size = image.size
        let msg = EMMessage.init(conversationID: receiver, from: sender, to: receiver, body: body, ext: ext)
        msg!.chatType = chatType
        return msg!
    }
    
    class func createVoiceMessage(_ localPath: String, _ displayName: String, _ duration: Int, to receiver: String, _  chatType: EMChatType, _ ext: Dictionary<String, Any>?) -> EMMessage {
        let body = EMVoiceMessageBody.init(localPath: localPath, displayName: displayName)
        if duration > 0 {
            body?.duration = Int32(duration)
        }
        let msg = EMMessage.init(conversationID: receiver, from: sender, to: receiver, body: body, ext: ext)
        msg!.chatType = chatType
        return msg!
    }

    class func createVideoMessage(_ localPath: String, _ displayName: String, _ duration: Int, to receiver: String, _  chatType: EMChatType, _ ext: Dictionary<String, Any>?) -> EMMessage {
        let body = EMVideoMessageBody.init(localPath: localPath, displayName: displayName)
        body?.duration = Int32(duration);
        let msg = EMMessage.init(conversationID: receiver, from: sender, to: receiver, body: body, ext: ext)
        msg!.chatType = chatType
        return msg!
    }

}
